modules/remotebackend/remotebackend.cc

```diff
@@ -889,7 +899,7 @@ bool RemoteBackend::searchRecords(const string& pattern, size_t maxResults, vect
     rr.qclass = QClass::IN;
     rr.content = stringFromJson(row, "content");
     rr.ttl = row["ttl"].int_value();
-    rr.domain_id = intFromJson(row, "domain_id", -1);
+    rr.domain_id = static_cast<domainid_t>(intFromJson(row, "domain_id", UnknownDomainID));
```

Comment:
I wondered if we should make domainidFromJson, but it's only in three spots, so not really necessary